frontend/src/Map/Markers.js

```diff
@@ -1,15 +1,20 @@
 import React from 'react';
 import MarkerWithPopup from './MarkerWithPopup.js';
 
-export default function Markers({ stations }) {
+export default function Markers({ stations, weatherIcons }) {
   return (
     <div>
       {stations
         .filter(
           (station) => station.latitude != null && station.longitude != null
         )
-        .map((station) => (
-          <MarkerWithPopup key={station.id} station={station}></MarkerWithPopup>
+        .map((station, index) => (
+          <MarkerWithPopup
+            weatherIcons={weatherIcons}
```

Comment:
Szerintem jobb lenne, ha nem raknál össze egy tömböt, amit index-szel együtt lepasszolgatsz végig, hanem a station-be raknál egy ikont, ami őhozzá tartozik. Ennek a számolása lehet itt vagy egy szinttel feljebb is akár.

Reply:
Javítva